temporalnexus/operation.go

```diff
@@ -324,12 +327,21 @@ func ExecuteUntypedWorkflow[R any](
 	if startWorkflowOptions.TaskQueue == "" {
 		startWorkflowOptions.TaskQueue = nctx.TaskQueue
 	}
+	if startWorkflowOptions.ID == "" {
+		return nil, internal.ErrMissingWorkflowID
+	}
 
 	if nexusOptions.RequestID != "" {
 		internal.SetRequestIDOnStartWorkflowOptions(&startWorkflowOptions, nexusOptions.RequestID)
 	}
 
 	if nexusOptions.CallbackURL != "" {
+		if nexusOptions.CallbackHeader == nil {
+			nexusOptions.CallbackHeader = make(nexus.Header)
+		}
+		if _, set := nexusOptions.CallbackHeader[nexus.HeaderOperationID]; !set {
```

Comment:
nit: should we use the case insensitive operations on the headers? I guess `nexus.HeaderOperationID` is now lower case so it isn't that important.